hermes-api/src/main/java/pl/allegro/tech/hermes/api/Subscription.java

```diff
@@ -35,6 +35,15 @@
     private EndpointAddress endpoint;
 
     @NotNull
+    private AuthenticationType authenticationType = AuthenticationType.NONE;
```

Comment:
I think it would be a good idea to merge all authentication related fields into `AuthenticationDetails` or something similar...